src/main/java/timify/com/studytime/StudyTimeService.java

```diff
@@ -0,0 +1,232 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import timify.com.common.apiPayload.exception.handler.StudyHandler;
+import timify.com.common.apiPayload.exception.handler.StudyTimeHandler;
+import timify.com.common.apiPayload.exception.handler.TodoHandler;
+import timify.com.member.domain.Member;
+import timify.com.studytime.domain.StudyTime;
+import timify.com.studytime.domain.StudyTimeGrade;
+import timify.com.studytime.dto.StudyTimeRequest.studyTimeRequest;
+import timify.com.studytime.repository.StudyTimeRepository;
+import timify.com.todo.TodoConverter;
+import timify.com.todo.domain.Todo;
+import timify.com.todo.dto.TodoRequest.todoRequest;
+import timify.com.todo.repository.TodoRepository;
+
+@Service
+@RequiredArgsConstructor
+public class StudyTimeService {
+
+    private final static double DEFAULT_TEMP = 50d;
+    private final StudyTimeRepository studyTimeRepository;
+    private final TodoRepository todoRepository;
+
+    @Transactional
+    public List<StudyTime> recordStudyTime(Member member, Long todoId, studyTimeRequest request) {
+        Todo todo = validateTodoOwner(member, todoId);
+
+        LocalDateTime startTime = stringToLocalTime(request.getStartTime());
+        LocalDateTime endTime = stringToLocalTime(request.getEndTime());
+
+        // 중복되는 시간 확인
+        List<StudyTime> overlappingTimes = studyTimeRepository.findByTodoAndStartTimeLessThanEqualAndEndTimeGreaterThanEqual(
+            todo, endTime, startTime);
+        if (!overlappingTimes.isEmpty()) {
+            throw new StudyTimeHandler(OVERLAP_STUDY_TIME);
+        }
+
+        List<StudyTime> studyTimes = new ArrayList<>();
+
+        // 4 AM 기준 시간 설정
+        LocalTime boundaryTime = LocalTime.of(4, 0);
+        LocalDateTime boundaryDateTime = LocalDateTime.of(startTime.toLocalDate(), boundaryTime);
+
+        if (endTime.isBefore(startTime)) {
+            throw new StudyTimeHandler(NOT_POSSIBLE_STUDY_TIME);
+        }
+
+        // 4 AM 넘는 경우
+        if (isCrossingBoundary(startTime, endTime, boundaryDateTime)) {
+            studyTimes.add(saveStudyTimePart(member, todo, startTime, boundaryDateTime, request));
+            studyTimes.add(
+                saveStudyTimePartForNextDay(member, todo, boundaryDateTime, endTime, request));
+            return studyTimes;
+        }
+
+        studyTimes.add(saveStudyTime(member, todo, startTime, endTime, request));
+        return studyTimes;
+    }
+
+    @Transactional
+    public void deleteStudyTime(Member member, Long studyTimeId) {
+
+        StudyTime studyTime = validateStudyTimeOwner(member, studyTimeId);
+
+        studyTime.disassociateMember(member);
+        studyTime.disassociateSubject(studyTime.getTodo().getSubject());
+        studyTime.disassociateTodo(studyTime.getTodo());
+        studyTimeRepository.delete(studyTime);
+
+        if (studyTimeRepository.countByTodo(studyTime.getTodo()) == 1) {
```

Comment:
여기도 repository를 이용하지 않아도 될 것 같아요!
```suggestion
        studyTime.getTodo().getStudyTimeList().size();
```